src/main/java/io/github/ryotan/code/util/CodeEnumReflectionUtil.java

```diff
@@ -0,0 +1,290 @@
+/*
+ * These codes are licensed under CC0.
+ *
+ * http://creativecommons.org/publicdomain/zero/1.0/deed
+ * http://creativecommons.org/publicdomain/zero/1.0/deed.ja
+ */
+
+package io.github.ryotan.code.util;
+
+import java.lang.annotation.Annotation;
+import java.lang.reflect.AnnotatedElement;
+import java.lang.reflect.Field;
+import java.lang.reflect.InvocationTargetException;
+import java.lang.reflect.Method;
+import java.lang.reflect.ParameterizedType;
+import java.lang.reflect.Type;
+import java.util.Optional;
+import java.util.function.Predicate;
+import java.util.stream.Stream;
+
+import io.github.ryotan.code.CodeEnum;
+import io.github.ryotan.code.CodeEnum.Filter;
+import io.github.ryotan.code.CodeEnum.ShortLabel;
+
+
+/**
+ * {@link CodeEnum}のリフレクション関連のユーティリティクラスです。
+ * <p>
+ * パラメータとして与えられたクラスが{@link CodeEnum}型のクラスである場合、コードパターンや短縮論理名を取得します。
+ * </p>
+ * <p>
+ * {@link Class}を引数に取るものは、与えられた{@link Class}がCodeEnum型でない場合には、{@link IllegalArgumentException}を送出します。
+ * </p>
+ *
+ */
+public final class CodeEnumReflectionUtil {
+    
+    /**
+     * hidden constructor for utility class.
+     */
+   private CodeEnumReflectionUtil() {
+    }
+
+    /**
+     * {@link Enum}かつ{@link CodeEnum}のタイプパラメータが{@code aClass}と一致する場合{@code true}を返却します。
+     * 
+     * @param aClass 判定対象のクラス
+     * @return タイプパラメータが{@code aClass}と一致する場合{@code true}、{@code aClass}が{@link Enum}または{@link CodeEnum}を継承していない場合{@code false}
+     */
+    public static boolean isValidCodeEnumClass(Class<?> aClass) {
+        if (!Enum.class.isAssignableFrom(aClass) || !CodeEnum.class.isAssignableFrom(aClass)) {
+            return false;
+        }
+
+        return Stream.of(aClass.getGenericInterfaces()).filter(ParameterizedType.class::isInstance).map(ParameterizedType.class::cast)
+                .filter(CodeEnumReflectionUtil::isCodeEnumType).anyMatch(type -> hasSameParameterizedType(type, aClass));
+    }
+
+    /**
+     * {@link CodeEnum}の実装クラスを返却します。
+     * <p>
+     * {@code code}がEnum型でない場合は、{@link IllegalArgumentException}を送出します。
+     * </p>
+     * 
+     * @param code コード値を表すクラス
+     * @return {@code CodeEnum}の実装クラス
+     * 
+     * @throws IllegalArgumentException {@code code}がEnum型でない場合
+     */
+    @SuppressWarnings("unchecked")
+    public static <C extends CodeEnum<C>> Class<C> getCodeEnumClass(Class<?> code) {
```

Comment:
![Codacy](https://www.codacy.com/assets/images/favicon.png) Issue found: [Expected @param tag for '<C>'.](https://www.codacy.com/app/ryotan/code-enum-support/file/3194383412/issues/source?bid=3375378&fileBranchId=3375797#l71)